Update `getting-started.md` to reflect changes for SRI 1.5.0

lucasbalieiro opened the pull request:
To be merged AFTER release of the SRI 1.5.0

This update a few git commands and add a new session explaining how to setup the aggregation modes on the translator proxy

lucasbalieiro commented on app/src/_developers/getting-started.md:
````diff
@@ -82,8 +82,113 @@ On the CPUMiner directory:
+#### Clone the repository
+
+```bash
+git clone https://github.com/stratum-mining/stratum.git
````
Isn't after the 1.6.0?

Because in the 1.5.0 we will still be in the older structure.

And the tutorial is trying to point to the latest stable release.

GitGab19 commented on app/src/_developers/getting-started.md:
We need to adjust lines 81-83.

There's no `channel_nominal_hashrate` parameter anymore.

Also, we need to tell the user to add their username in the `user_identity` field of `jdc-config-hosted-example.toml` file. I would also add this suggestion in the paragraph related to optional adjustments.

GitGab19 commented on app/src/_developers/getting-started.md:
We also need to adjust lines 19-22 accordingly

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -82,8 +82,113 @@ On the CPUMiner directory:
+
+In Stratum V2, a **channel** represents a dedicated mining session between a downstream mining device and an upstream node (such as a pool, job declarator, or proxy). Each channel has its own mining target, accepts work submissions, and is uniquely identified by a `channel_id`.
+
+You can configure whether the Translator Proxy (TProxy) manages channels in **aggregated** or **non-aggregated** mode using the `aggregate_channels` parameter in `tproxy-config-local-jdc-example.toml`.
```
```suggestion
You can configure whether the Translator Proxy (TProxy) manages channels in **aggregated** or **non-aggregated** mode using the `aggregate_channels` parameter in `tproxy-config-local-pool-example.toml`.
```

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -82,8 +82,113 @@ On the CPUMiner directory:
+You can configure whether the Translator Proxy (TProxy) manages channels in **aggregated** or **non-aggregated** mode using the `aggregate_channels` parameter in `tproxy-config-local-jdc-example.toml`.
+
+* **Non-Aggregated (default: `false`)** – Each mining device connected to the TProxy opens its own upstream channel. This provides isolation per device and is generally the preferred setup.
+* **Aggregated (`true`)** – All miners connected to the TProxy share a single upstream channel. This mode exists mainly for compatibility with Stratum V1-style proxies but can also reduce network overhead in certain setups.
```
```suggestion
* **Aggregated (aggregate_channels: `true`)** – All miners connected to the Translator Proxy share a single upstream channel. This mode reduces significantly the bandwidth usage, but doesn't let the Pool to have monitoring capabilities on individual mining devices.
```

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -82,8 +82,113 @@ On the CPUMiner directory:
+
+For a detailed explanation of channels, see the [Stratum V2 specification, Section 5.2](https://stratumprotocol.org/specification/05-Mining-Protocol/#52-channel).
+
+## III Getting Started - Running all roles 
```
```suggestion
## III Getting Started - Running all roles with Job Declaration (JD)
```

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -82,8 +82,113 @@ On the CPUMiner directory:
+
+* **Non-Aggregated (default: `false`)** – Each mining device connected to the TProxy opens its own upstream channel. This provides isolation per device and is generally the preferred setup.
+* **Aggregated (`true`)** – All miners connected to the TProxy share a single upstream channel. This mode exists mainly for compatibility with Stratum V1-style proxies but can also reduce network overhead in certain setups.
+
```
Here we need the paragraph related to optional adjustments where we tell the user to set the `user_identity` and `min_individual_miner_hashrate` in the config files.

lucasbalieiro commented:
@GitGab19, I’ve addressed your points.

I also updated the section *“Monitor for mined blocks on Testnet”* since the links were pointing to **Testnet3**, while the tutorial sets up a **Testnet4** node.
Additionally, I removed the Blockstream link, as it only provided information for **Testnet3**.

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -80,10 +81,125 @@ On the CPUMiner directory:
 #### Adjust proxy-config (optional)
 
-Depending on mining device you do run, you may have to adjust `tproxy-config-local-jdc-example.toml` file in order to adjust the `min_individual_miner_hashrate` and `channel_nominal_hashrate` parameters accordingly
+Depending on mining device you do run, you may have to adjust `tproxy-config-local-jdc-example.toml` file in order to adjust the `min_individual_miner_hashrate` parameter accordingly.
```
We are missing the `user_identity` modification on JDC config file here.

GitGab19 replied:
I would add the section as you did later.

lucasbalieiro replied:
Section added to the three setups, adapted accordingly

GitGab19 commented:
Let's add the `Translator Proxy configurations (optional)` paragraph also in the JD configuration.

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -190,7 +324,29 @@ Setup the correct CPUMiner for your OS:
+
+#### Translator Proxy and Job Declarator configurations (optional)
+
+##### - Aggregated vs. Non-Aggregated mode
+
+In Stratum V2, a **channel** represents a dedicated mining session between a downstream mining device and an upstream node (such as a pool, job declarator, or proxy). Each channel has its own mining target, accepts work submissions, and is uniquely identified by a `channel_id`.
+
+You can configure whether the Translator Proxy (TProxy) manages channels in **aggregated** or **non-aggregated** mode using the `aggregate_channels` parameter in `tproxy-config-local-jdc-example.toml`.
+
+* **Non-Aggregated (aggregate_channels: `false`)** – Each mining device connected to the Translator Proxy opens its own upstream channel. This provides isolation per device and let the Pool to have monitoring capabilities on individual mining devices.
+* **Aggregated (aggregate_channels: `true`)** – All miners connected to the Translator Proxy share a single upstream channel. This mode reduces significantly the bandwidth usage, but doesn't let the Pool to have monitoring capabilities on individual mining devices.
+
+For a detailed explanation of channels, see the [Stratum V2 specification, Section 5.2](https://stratumprotocol.org/specification/05-Mining-Protocol/#52-channel).
```
This is not needed in the JD scenario, remove it

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -190,7 +324,29 @@ Setup the correct CPUMiner for your OS:
+
+##### - User Identity
+
+You may also want to adjust the `user_identity` parameter in `tproxy-config-local-jdc-example.toml` and `jdc-config-local-example.toml`. This parameter defines an identifier that will be visible to the upstream apps. A counter is automatically appended for each connected client (e.g., `username.miner1`, `username.miner2`).
```
```suggestion
You may also want to adjust the `user_identity` parameter in `tproxy-config-local-jdc-example.toml` and `jdc-config-local-example.toml`.
```

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -78,12 +79,145 @@ On the CPUMiner directory:
 
-Depending on mining device you do run, you may have to adjust `tproxy-config-local-jdc-example.toml` file in order to adjust the `min_individual_miner_hashrate` and `channel_nominal_hashrate` parameters accordingly
+##### - Aggregated vs. Non-Aggregated mode
 
+In Stratum V2, a **channel** represents a dedicated mining session between a downstream mining device and an upstream node (such as a pool, job declarator, or proxy). Each channel has its own mining target, accepts work submissions, and is uniquely identified by a `channel_id`.
 
-## II Getting Started - Running all roles 
+You can configure whether the Translator Proxy (TProxy) manages channels in **aggregated** or **non-aggregated** mode using the `aggregate_channels` parameter in `tproxy-config-local-jdc-example.toml`.
+
+* **Non-Aggregated (aggregate_channels: `false`)** – Each mining device connected to the Translator Proxy opens its own upstream channel. This provides isolation per device and let the Pool to have monitoring capabilities on individual mining devices.
+* **Aggregated (aggregate_channels: `true`)** – All miners connected to the Translator Proxy share a single upstream channel. This mode reduces significantly the bandwidth usage, but doesn't let the Pool to have monitoring capabilities on individual mining devices.
+
+For a detailed explanation of channels, see the [Stratum V2 specification, Section 5.2](https://stratumprotocol.org/specification/05-Mining-Protocol/#52-channel).
```
No need for this since here we are doing JD

GitGab19 reviewed (dismissed):
ACK, thanks @lucasbalieiro!

plebhash commented:
when I render https://github.com/lucasbalieiro/stratumprotocol.org/blob/update-getting-started-guid-for-1.5.0/app/src/_developers/getting-started.md I cannot see the images on the `IV Final Step: monitoring for blocks` section

I can see them here: https://stratumprotocol.org/developers#iii-final-step-monitoring-for-blocks

but those logs are outdated

plebhash commented:
I was looking at the renderization on github, I guess that's just markdown syntax that takes relative paths

end of the day, the code isn't written to be rendered on github, so I guess that's not a major problem

but yeah, it would be good to have updated screenshots

lucasbalieiro commented:
Updated with more recent screenshots of a block found
